### DISCUSSION TITLE
Business rules

### DISCUSSION THREAD

**Discussion by oskarjf** [MarimerLLC/csla#3282] [Questions] [answered]
I'm working on CSLA v.5.5.3
I'm struggling to figure out exactly what to have in my Execute method when validating my BL library. 
I've been looking at the samples in RuleTutorial.
Is it enough to add an error message to the rule context with context.AddErrorResult()?  
Should I normally just add the property to the out values with context.AddOutValue if validation succeeds?

**Comment by oskarjf** [comment]
For some of the example rules it's returning
`context.Results.Add(new RuleResult(this.RuleName, field.Key, string.Format(GetMessage(), fieldNames)) { Severity = this.Severity });`
on failure. How do I test that in a Unit Test?

**Comment by Chicagoan2016** [comment, marked as answer]
I put my 'main' logic in Execute method. Some of the following are examples.
1) To make sure End Date is later than Start Date, the comparison is in Execute method.
2) Calculate amount of 'tax' for item types. This in turn calls a command object (inside Execute method).
3) Certain fields are required only when another property in object is 'true'.

The default is context.AddSuccessResult() but majority of our business rules call context.AddErrorResult()

context.AddOutValue is only used when you want to set value of your business property from within your business rule.

In our projects, I use a Resource file to get 'messages' for my business rules.

**Reply by oskarjf** [reply]
Thanks for the reply :)
Do you unit test a certain rule or just that the object that the rules apply to is valid?

**Reply by Chicagoan2016** [reply]
In the Rule Tutorial sample, you could look at the test code for  testing rules but I only test my objects : ). Laziness!!!
